Document System Variables usage for Windows OS

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Referring to: https://github.com/checkstyle/checkstyle/pull/15028#discussion_r1647559846

In this PR, I had modified the `severity` property of [google_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml) from:
```
<property name="severity" value="warning"/>
```

to

```
<property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
```

So we should be able to dynamically change severity level of google_checks.xml using system variable in CLI by something like:

```
java -Dorg.checkstyle.google.severity=error -jar checkstyle-xx.xx.x-all.jar -c google_checks.xml Test.java
```

But this gives error. Here is an example of it:

#### Test.java
```java
public class Test {}
```

#### Output
```
$ java -Dorg.checkstyle.google.severity=error -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java
Error: Could not find or load main class .checkstyle.google.severity=error
Caused by: java.lang.ClassNotFoundException: /checkstyle/google/severity=error
```

Now if I modify the `severity` property in google_checks.xml to something like:

```
<property name="severity" value="${testing}" default="warning"/>
```

This works fine. Here's an example of it:

```
$ java -Dtesting=error -jar .\checkstyle-10.17.0-all.jar -c .\google_checks.xml .\Test.java                       
Starting audit...
[ERROR] C:\checkstyle testing\TestingGoogleStyle\.\Test.java:1:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

The CLI is working perfectly for this modification, we're able to change severity level from `warning` to `error`. 

We should expect the same behavior for `severity` property mentioned at the beginning:

```
<property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>
```

Proof issue above looks to be a Windows issue, and that the new property works via system property:
````
PS \target> java -jar .\checkstyle-10.18.0-SNAPSHOT-all.jar -c google_checks.xml .\Test.java
Starting audit...
[WARN] M:\checkstyleWorkspaceEclipse\checkstyle\target\.\Test.java:1:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.

PS \target> java -Dorg.checkstyle.google.severity=error -jar .\checkstyle-10.18.0-SNAPSHOT-all.jar -c google_checks.xml .\Test.java
Error: Could not find or load main class .checkstyle.google.severity=error
Caused by: java.lang.ClassNotFoundException: /checkstyle/google/severity=error

PS \target> java -D"org.checkstyle.google.severity=error" -jar .\checkstyle-10.18.0-SNAPSHOT-all.jar -c google_checks.xml .\Test.java
Starting audit...
[ERROR] M:\checkstyleWorkspaceEclipse\checkstyle\target\.\Test.java:1:1: Missing a Javadoc comment. [MissingJavadocType]
Audit done.
Checkstyle ends with 1 errors.
````

I am closing this issue.

--- a/src/xdocs/cmdline.xml.vm
+++ b/src/xdocs/cmdline.xml.vm
@@ -148,7 +148,10 @@ java -D&lt;property&gt;=&lt;value&gt;  \
         option has been dropped for Checkstyle 5.0, because of significant changes regarding
         package name file handling. See <a href="config.html#Packages">this link</a> for details.
       </p>
-
+      <p>
+        To define system property for execution look at
+        <a href="config_system_properties.html#Properties">system properties</a>.
+      </p>
       <p>
         Set the properties for <a href="config.html#Properties">expanded
         property values</a> by either by assigning system properties using the

--- a/src/xdocs/config_system_properties.xml
+++ b/src/xdocs/config_system_properties.xml
@@ -16,6 +16,17 @@
       </macro>
     </section>
 
+    <section name="How to set system property for CLI">
+      <p>
+        System properties should be provided to <code>java</code> command.
+        Example: <code>java -D&lt;property&gt;=&lt;value&gt; -jar ... </code>
+      </p>
+      <p>
+        Attention to Windows users: system properties should be wrapped with <code>"</code>.
+        Example: <code>-D"&lt;property&gt;=&lt;value&gt;"</code>
+      </p>
+    </section>
+
     <section name="Enable External DTD load">
       <p>
         The property <code>checkstyle.enableExternalDtdLoad</code>
@@ -118,6 +129,10 @@ Starting audit...
 Audit done.
 Checkstyle ends with 2 errors.
           </source>
+        </p>
+        <p>
+          Attention to Windows users: system properties should be wrapped with <code>"</code>.
+          Example: <code>-D"&lt;property&gt;=&lt;value&gt;"</code>
         </p>
       </subsection>
     </section>

